test/test-deploy-endpoints.js

```diff
@@ -33,36 +34,62 @@ test('Create and destroy a service', function(t) {
   }
   TestServiceManager.prototype.getDeployment = getDeployment;
 
-  t.plan(16);
   var server = meshServer(new TestServiceManager());
-  server.set('port', 0);
-  server.start(function(err, port) {
-    t.ok(!err, 'Server should start');
+  var client = null;
+  var service = null;
 
-    var client = new Client('http://127.0.0.1:' + port + '/api');
-    client.serviceCreate('My Service', 2, function(err, service) {
-      t.ok(!err, 'Create service should succeed');
-      t.equal(service.name, 'My Service', 'Service name should match');
-      t.equal(service._groups.length, 1, 'Service should have 1 group');
-      t.equal(service._groups[0].scale, 2, 'Group scale should be 2');
-      var req = client.serviceDeploy(
-        service,
-        'application/some-type',
-        function(err, res) {
-          t.ok(!err, 'Deploy should not error');
-          var responseBody = 'ok. ServiceID: ' + service.id;
-          t.equal(res.body, responseBody, 'deploy: Response should match');
+  t.test('start server', function(tt) {
+    server.set('port', 0);
+    server.start(function(err, port) {
+      tt.ifError(err);
+      client = new Client('http://127.0.0.1:' + port + '/api');
+      tt.end();
+    });
+  });
 
-          client.serviceGetArtifact(service, function(err, res) {
-            t.ok(!err, 'download: Deployment download should succeed');
-            var expected = 'Data. ServiceID: ' + service.id;
-            t.equal(res.body, expected, 'download: Download body should match');
-            server.stop();
-          });
-        }
-      );
+  t.test('inialize models', function(tt) {
+    client.serviceCreate('My Service', 2, function(err, s) {
+      tt.ifError(err);
+      tt.equal(s.name, 'My Service', 'Service name should match');
+      tt.equal(s._groups.length, 1, 'Service should have 1 group');
+      tt.equal(s._groups[0].scale, 2, 'Group scale should be 2');
+      service = s;
+      tt.end();
+    });
+  });
 
-      req.write('some data');
+  t.test('deploy a service', function(tt) {
+    var req = service.deploy('application/some-type', function(err, res) {
+      tt.ifError(err);
+      var expResponseBody = 'ok. ServiceID: ' + service.id;
+      res.setEncoding('utf8');
+      res.on('error', t.ifError.bind(t));
```

Comment:
Hm.. as convenient as it is, binding is a one-way street and I'd be worried that people are depending on being able to do the binding themselves. It would also interfere with some extension patterns.

How about a tap wrapper that does [lodash.bindAll()](https://lodash.com/docs#bindAll) on `t` before passing it to your function? Or even just `t = _.bindAll(t);` on the first line is pretty unobtrusive.